Fix shade of 3-button navigation bar in HomeFragment on app start

min7-i opened the pull request:
In #1622 I missed an issue where the 3-button navigation bar in the HomeFragments stays transparent on app start. This is caused by [isButtonNavigation()](https://github.com/breezy-weather/breezy-weather/blob/96b82d2403e566e639fbf0feaaa43f3680e250dc/app/src/main/java/org/breezyweather/ui/main/fragments/HomeFragment.kt#L121C17-L121C35) which is called before the root view - used in [setSystemBarStyle()](https://github.com/breezy-weather/breezy-weather/blob/96b82d2403e566e639fbf0feaaa43f3680e250dc/app/src/main/java/org/breezyweather/ui/main/fragments/HomeFragment.kt#L155) to determine if a button navigation bar is present - is attached. As a result, the method always returns `false` and no shade is applied to the navigation bar. This didn't occur before because [checkToSetSystemBarStyle()](https://github.com/breezy-weather/breezy-weather/blob/96b82d2403e566e639fbf0feaaa43f3680e250dc/app/src/main/java/org/breezyweather/ui/main/fragments/HomeFragment.kt#L547) and thus `setSystemBarStyle` was called multiple times on app start which is not the case anymore.

However, I realized that with the changes made in #1457 `isButtonNavigation` is no longer needed. When [window.isNavigationBarContrastEnforced](https://github.com/breezy-weather/breezy-weather/blob/96b82d2403e566e639fbf0feaaa43f3680e250dc/app/src/main/java/org/breezyweather/common/extensions/DisplayExtensions.kt#L152) is `true`, it automatically determines if the navigation bar should be shaded. That way the navigation bar is shaded when 3-button navigation or a taskbar (on tablets) is used while keeping the navigation bar transparent when gesture navigation is active.

Thoroughly tested on the following devices with both navigation types:
- [x] Google Pixel 6a (Android 15)
- [x] emulator with Android 9 [only 3-button navigation available]
- [x] Galaxy S25 (Android 15) [via Samsung Remote Test Lab]
- [x] Galaxy A54 (Android 14) [via Samsung Remote Test Lab]
- [x] Galaxy Tab S9 (Android 14) [via Samsung Remote Test Lab]
- [x] emulator with Android 7 [only 3-button navigation and dark navigation bar available]

papjul commented:
In that case, I think you can remove the `navigationShader` parameter in cascade, since it’s always `true` now.

min7-i commented:
You're right. I removed the parameter and simplified `setSystemBarStyle` a bit.

If we ever need to make the navigation bar transparent, we can still get the logic back from the commit history.

papjul commented:
Thank you!
